CPP-Example.cpp

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+   char src[40];
+   char dest[100];
+   char dest2[100];
+
+   gets(src); 
+
+   strcpy(dest, src); 
+ 
+   strncpy(dest,src, strlen(dest)); //DevSkim: ignore DS111237,DS154189 
+   
+   strcat(dest, "this is bad");
+
+   strncat(dest, "this is also bad", strlen(dest)); 
+
+   memcpy(dest2, dest, sizeof dest);
```

Comment:
[Problematic C function detected (memcpy)](https://github.com/Microsoft/DevSkim/blob/master/guidance/DS121708.md)
There are a number of conditions in which memcpy can introduce a vulnerability (mismatched buffer sizes, null pointers, etc.). More secure alternitives perform additional validation of the source and destination buffer
Suggestions:

_+   memcpy_s(dest2, <size of dest2>,  dest,  sizeof dest);_